docs/concepts.rst

```diff
@@ -116,6 +116,47 @@ DAGs can be used as context managers to automatically assign new operators to th
 
     op.dag is dag # True
 
+.. _concepts:functional_dags:
+
+Functional DAGs
+---------------
+*Added in Airflow 1.10.11*
+
+DAGs can be defined using functional abstractions. Outputs and inputs are sent between tasks using
+:ref:`XComs <concepts:xcom>` values. In addition, you can wrap functions as tasks using the
+:ref:`task decorator <concepts:task_decorator>`. Dependencies are automatically inferred from
+the message dependencies.
+
+Example DAG with functional abstraction
+
+.. code:: python
+
+  with DAG(
+      'send_server_ip', default_args=default_args, schedule_interval=None
+  ) as dag:
+
+    # Using default connection as it's set to httpbin.org by default
+    get_ip = SimpleHttpOperator(
+        task_id='get_ip', endpoint='get', method='GET', xcom_push=True
+    )
+
+    @dag.task(multiple_outputs=True)
+    def prepare_email(raw_json: str) -> str:
+      external_ip = json.loads(raw_json)['origin']
+      return {
+        'subject':f'Server connected from {external_ip}',
+        'body': f'Seems like today your server executing Airflow is connected from the external IP {external_ip}<br>'
```

Comment:
Unfortunately we can't use f-Strings, because we still support Python 2.7 in the 1.10.x series.

Reply:
So if we want this in 1.10.11 we need to be very careful :/

Reply:
The question is do we want to backport it?

Reply:
Or we could use [future-fstrings](https://github.com/asottile/future-fstrings). We already have it as dep of a [dep](https://github.com/apache/airflow/search?q=future-fstrings&unscoped_q=future-fstrings) and add `# -*- coding: future_fstrings -*-`.

Reply:
Given this is an example DAG on a document, I don't think it really matters that much. If users want to use it on python 2 (which btw is already deprecated) they can change it to format. But I think we should adapt to use py3 examples better.

Reply:
There are more cases where you are using F-Strings https://github.com/apache/airflow/pull/8962#discussion_r431670798

Reply:
f-strings are not a blocker and we are using them in many places